core/GlobalState.cc

```diff
@@ -1412,6 +1418,72 @@ bool GlobalState::wasModified() const {
     return wasModified_;
 }
 
+bool GlobalState::isTypecheckingCanceled() const {
+    return lspEpochInvalidator->load() != currentlyProcessingLSPEpoch->load();
+}
+
+void GlobalState::startCommitEpoch(u4 epoch) {
+    absl::MutexLock lock(epochMutex.get());
```

Comment:
is there a relationship between `epoch` and `currentlyProcessingLSPEpoch` that we can enforce?
e.g. is one different from other in +1 or can it be arbitrary big difference?

(please leave answer as a comment in source)